homeassistant/components/config/entity_registry.py

```diff
@@ -42,18 +54,16 @@ def websocket_list_entities(
         msg: dict[str, Any],
     ) -> None:
         """Handle list registry entries command."""
-        nonlocal cached_list_entities
-        if not cached_list_entities:
-            registry = er.async_get(hass)
-            cached_list_entities = message_to_json(
-                websocket_api.result_message(
-                    IDEN_TEMPLATE,  # type: ignore[arg-type]
-                    [_entry_dict(entry) for entry in registry.entities.values()],
-                )
-            )
-        connection.send_message(
-            cached_list_entities.replace(IDEN_JSON_TEMPLATE, str(msg["id"]), 1)
+        registry = er.async_get(hass)
+
+        result = ",".join(
+            _async_get_cached_entity(entity_id, entry)
```

Comment:
You could speed this up a bit more by not making this a function as the python function overhead for 1000s of entities likely has a performance impact here